### ISSUE TITLE
Ignoring deletions, pending deletions even all endpoints are in sync

### ISSUE THREAD

**Issue by pug77** [syncthing/syncthing#8245]
Syncthing 1.9.1

**Issue:**
syncthing is showing out-of-sync items (pending deletions) when some endpoint is ignoring deletions even everything is in sync

**Expected:**
deletion should be always accepted if object (file/folder) was already deleted on receiving endpoint

**Example:**
A endpoint (send-receive)
B endpoint (send-receive and ignoring deletions)

1. A & B are synced
2. A will delete file XY
3. B is out of sync (ignores deletion) OK
4. B will delete same file XY
5. B is still out of sync NOT OK (but deletion should be accepted because file does not exists on B anymore)

**Comment by calmh** [comment]
We don't like ignore-deletes very much and trouble there is kind of expected, but in this case I think there's a more general point about merging file states that are "different" but equivalent. Seems legit until further investigation proves otherwise. :)

**Comment by king0billy** [comment]
how to pend deletions on the receiver when the sender deletes many files in the sync-folder?

**Comment by cahalu** [comment]
I have same issue with https://github.com/Catfriend1/syncthing-android in version **1.29.2**. In app there are pending deletions and it is displayed out of sync **95%** 0B. Files and directories doesn't exists, but it should be deleted again. I try to fix it by delete deltas but without effect. Only working solution is to create files in source device and delete it again, but it is laborious.

Related:
https://github.com/Catfriend1/syncthing-android/issues/607
https://github.com/Catfriend1/syncthing-android/issues/863

**Comment by tomasz1986** [comment]
@cahalu It doesn't sound like you're using `ignoreDelete`, which is the culprit in this issue. However, does your problem look similar to https://forum.syncthing.net/t/deleted-files-randomly-getting-stuck-in-sync/23668?

**Comment by cahalu** [comment]
Yes, I use model A ↔ B ↔ C and problem is only in B device exactly as is described in https://forum.syncthing.net/t/deleted-files-randomly-getting-stuck-in-sync/23668.
